lib/topological_inventory/sync/host_inventory_response_worker.rb

```diff
@@ -0,0 +1,160 @@
+          return
+        end
+
+        new_vm = TopologicalInventoryIngressApiClient::Vm.new(
+          :source_ref          => external_id,
+          :host_inventory_uuid => host["id"]
+        )
+
+        save_vms_to_topological_inventory([new_vm], source_id)
+        registered_messages.delete(external_id)
+        logger.info("Topological Inventory has been updated with '#{host['display_name']}' VM (source:'#{source_id}').")
+      rescue => e
+        logger.error("#{e.message} -  #{e.backtrace.join("\n")}")
+      end
+
+      def check_timeouts(threshold = config.response_timeout)
+        expired = []
+        #
+        # STEP 1 Collect expired messages
+        #
+        registered_messages.each do |external_id, msg|
+          timeout_lock.synchronize do
+            if msg[:created_at] < Time.now.utc - threshold
+              expired << external_id
+            end
+          end
+        end
+
+        #
+        # STEP 2 Remove expired messages, send timeout callbacks
+        #
+        expired.each do |external_id|
+          registered_messages.delete(external_id)
+          logger.debug("ResponseWorker: Registrated request for external_id = '#{external_id}' has reached the timeout and it has been removed.")
+        end
+
+        sleep(config.response_timeout_poll_time)
+      rescue => err
+        logger.error("Exception in maintenance worker: #{err}\n#{err.backtrace.join("\n")}")
+      end
+
+      def save_vms_to_topological_inventory(topological_inventory_vms, source)
+        # TODO(lsmola) if VM will have subcollections, this will need to send just partial data, otherwise all subcollections
+        # would get deleted. Alternative is having another endpoint than :vms, for doing update only operation. Partial data
+        # are not supported without timestamp for update (at least for now), since that is just being added to skeletal
+        # precreate.
+        TopologicalInventory::Providers::Common::SaveInventory::Saver.new(:client => ingress_api_client, :logger => logger).save(
+          :inventory => TopologicalInventoryIngressApiClient::Inventory.new(
+            :schema                     => TopologicalInventoryIngressApiClient::Schema.new(:name => "Default"),
+            :source                     => source,
+            :collections                => [
+              TopologicalInventoryIngressApiClient::InventoryCollection.new(:name => :vms, :data => topological_inventory_vms)
+            ],
+            :refresh_state_part_sent_at => Time.now.utc
+          )
+        )
+      end
+
+      def ingress_api_client
+        TopologicalInventoryIngressApiClient::DefaultApi.new
+      end
+
+      # No persist_ref here, because all instances (pods) needs to receive kafka message
+      def queue_opts
+        opts               = {:service  => config.queue_topic,
+                              :auto_ack => config.queue_auto_ack}
```

Comment:
Should we default to auto_ack if it is not set? like on line 145, 146?

Reply:
You are right. If there is no `:auto_ack` key it is used `true` as a default value. But in this case key would be present, so the `nil` would be used as a value ([#subscribe_topic_impl](https://github.com/ManageIQ/manageiq-messaging/blob/236300ec62608dd3d3c63f5c28b28560c014091c/lib/manageiq/messaging/kafka/topic.rb#L19) -> [#auto_ack?](https://github.com/ManageIQ/manageiq-messaging/blob/236300ec62608dd3d3c63f5c28b28560c014091c/lib/manageiq/messaging/common.rb#L40)).